### PR TITLE
*: enable range typed table partition

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Enable range typed table partition by default.

### What is changed and how it works?

If the SQL is `create table ... partition by range ...`, 
the `PartitionInfo.Enable` field will be set to true
So this kind of table partition is handled.

### Check List 

Tests 

 - No code

Side effects

 - Breaking backward compatibility

If a user create a range typed table partition with this TiDB binary, he can't degrade to an old version.

@shenli @ciscoxll

### CODE REVIEW

**Comment by ciscoxll** [comment]
LGTM

**Comment by tiancaiamao** [comment]
After a partitioned table is created, turn off the flag is to late:
there is no way to handle a partitioned table as a normal table because the data is already persisted.

So this feature is controlled by the data, rather than by the code, provide a flag can't help.
@jackysp

**Comment by jackysp** [comment]
For some POCs, the customers could rebuild the cluster and continue the POC test if they could disable the partition feature. Otherwise, the test may be blocked. @tiancaiamao

File: executor/show.go [hunk 1]
```diff
@@ -620,6 +620,7 @@ func (e *ShowExec) fetchShowCreateTable() error {
 	if partitionInfo != nil {
 		// this if statement takes care of range columns case
 		if partitionInfo.Columns != nil && partitionInfo.Type == model.PartitionTypeRange {
+
```
Comment (zimulala) [inline executor/show.go]: Why add this line?

**Comment by tiancaiamao** [comment]
Any idea about the error? @zimulala

**Pull request merged**